### PR TITLE
Double instead of Float in LoadDetail

### LINKED ISSUE #1968: Double instead of Float in LoadDetail
* **Do you want to request a *feature* or report a *bug*?**
Enhancement

* **What is the current behavior?**
`LoadDetail` fields (in `iidm-extensions`) are `float`.

* **What is the expected behavior?**
For precision purposes, we would like to use `double`.

* **Please tell us about your environment:**
  - PowSyBl Version: 4.7.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by bapan25**
Signed-off-by: Ruptam <ruptam25@outlook.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
Fix #1968 

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/extensions/LoadDetailXmlSerializer.java [hunk 1]
```diff
@@ -33,29 +33,29 @@ public LoadDetailXmlSerializer() {
...
-        if (Float.isNaN(fixedActivePower)) {
-            fixedActivePower = XmlUtil.readFloatAttribute(context.getReader(), "subLoad1ActivePower");
+        double fixedActivePower = XmlUtil.readOptionalDoubleAttribute(context.getReader(), "fixedActivePower");
+        if (Double.isNaN(fixedActivePower)) {
+            fixedActivePower = XmlUtil.readDoubleAttribute(context.getReader(), "subLoad1ActivePower");
         }
-        float fixedReactivePower = XmlUtil.readOptionalFloatAttribute(context.getReader(), "fixedReactivePower");
-        if (Float.isNaN(fixedReactivePower)) {
-            fixedReactivePower = XmlUtil.readFloatAttribute(context.getReader(), "subLoad1ReactivePower");
+        double fixedReactivePower = XmlUtil.readDoubleAttribute(context.getReader(), "fixedReactivePower");
```
Comment (miovd) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/extensions/LoadDetailXmlSerializer.java:48]: This should be `readOptionalDoubleAttribute` here

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/extensions/LoadDetailXmlSerializer.java [hunk 2]
```diff
@@ -33,29 +33,29 @@ public LoadDetailXmlSerializer() {
...
+        if (Double.isNaN(fixedReactivePower)) {
+            fixedReactivePower = XmlUtil.readDoubleAttribute(context.getReader(), "subLoad1ReactivePower");
         }
-        float variableActivePower = XmlUtil.readOptionalFloatAttribute(context.getReader(), "variableActivePower");
-        if (Float.isNaN(variableActivePower)) {
-            variableActivePower = XmlUtil.readFloatAttribute(context.getReader(), "subLoad2ActivePower");
+        double variableActivePower = XmlUtil.readDoubleAttribute(context.getReader(), "variableActivePower");
```
Comment (miovd) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/extensions/LoadDetailXmlSerializer.java:52]: This should be `readOptionalDoubleAttribute` here

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/extensions/LoadDetailXmlSerializer.java [hunk 3]
```diff
@@ -33,29 +33,29 @@ public LoadDetailXmlSerializer() {
...
+        if (Double.isNaN(variableActivePower)) {
+            variableActivePower = XmlUtil.readDoubleAttribute(context.getReader(), "subLoad2ActivePower");
         }
-        float variableReactivePower = XmlUtil.readOptionalFloatAttribute(context.getReader(), "variableReactivePower");
-        if (Float.isNaN(variableReactivePower)) {
-            variableReactivePower = XmlUtil.readFloatAttribute(context.getReader(), "subLoad2ReactivePower");
+        double variableReactivePower = XmlUtil.readDoubleAttribute(context.getReader(), "variableReactivePower");
```
Comment (miovd) [inline iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/extensions/LoadDetailXmlSerializer.java:56]: This should be `readOptionalDoubleAttribute` here

**Review by miovd — CHANGES_REQUESTED** [review]
Almost there! Some corrections are still to be done.

File: iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/extensions/LoadDetailXmlSerializer.java (hunk 1 above)
Comment (bapan25) [reply iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/extensions/LoadDetailXmlSerializer.java]: Used readOptionalDoubleAttribute before NaN checking

**Review by miovd — APPROVED** [review]

**Comment by miovd** [comment]
Thank you for your PR, it is ready to be merged for me!

However, Github Actions do not seem to work on a PR from a forked repository. We will try to solve this issue but in the time being, could you please reopen an identical PR from this repository? You were added as a developper of the organization and should have the rights to clone and push a new branch.

Sorry for the inconvenience.

**Comment by bapan25** [comment]
Hi @miovd 
Raised a new PR
https://github.com/powsybl/powsybl-core/pull/1973

Hence closing this PR.

**Comment by miovd** [comment]
> Hi @miovd Raised a new PR #1973
> 
> Hence closing this PR.

Thanks a lot!

**Pull request closed without merging**